src/base-command.ts

```diff
@@ -92,11 +99,98 @@ export default abstract class BaseCommand extends Command {
     return super.parse(options, [...args, ...flags]);
   }
 
+  async _getFilteredMetadata(calling_class: any): Promise<any> {
+    const { args, flags } = await this.parse(calling_class);
+    const _args = Object.entries(args).map(([key, value]) => {
+      if (calling_class.non_sensitive.has(key)) {
+        return { [key]: value };
+      } else if (calling_class.sensitive.has(key)) {
+        return { '[Filtered]': '**********' };
+      } else {
+        return { '[Unknown]': '**********' };
+      }
+    });
+
+    const _flags = Object.entries(flags).map(([key, value]) => {
+      if (calling_class.non_sensitive.has(key)) {
+        return { [key]: value };
+      } else if (calling_class.sensitive.has(key)) {
+        return { '[Filtered]': '**********' };
+      } else {
+        return { '[Unknown]': '**********' };
+      }
+    });
+
+    return { args: _args, flags: _flags };
+  }
+
+  async _logToSentry(err: any): Promise<void> {
+    const auth_result = await this.app.auth.getPersistedTokenJSON();
+    const auth_user = await this.app.auth.checkLogin();
+    const metadata = await this._getFilteredMetadata(this.constructor as any);
+
+    Sentry.init({
+      dsn: CLI_SENTRY_DSN,
+      debug: false,
+      attachStacktrace: true,
+      environment: this.config.bin,
+      integrations: [
+        new Dedupe(),
+        new RewriteFrames({
+          root: __dirname || process.cwd(),
+        }),
+        new ExtraErrorData(),
+        new Transaction(),
+      ],
+      beforeSend(event: any) {
+        // Prevent sending sensitive information like access tokens to sentry
+        if (event.req?.data?.token) {
+          event.req.data.token = '*'.repeat(20);
+        }
+        return event;
+      },
+      initialScope: {
+        user: {
+          email: auth_result?.email,
+          id: auth_user.id,
+        },
+        extra: {
+          ...this.config,
+          ...this.app.config.toSentry(),
+          command_args: metadata.args,
+          command_flags: metadata.flags,
+          config_file: path.join(this.app.config.getConfigDir(), LocalPaths.CLI_CONFIG_FILENAME),
+          cwd: process.cwd(),
+          docker_info: execSync('docker version').toString(),
```

Comment:
Might want to just use the `docker` util here instead

Reply:
refactored using docker